woodchipper/http/fastapi.py

```diff
@@ -64,7 +64,7 @@ async def send_with_extra_headers(message):
                 "id": self._request_id_factory(),
                 "body_size": int(request.headers.get("content-length", 0)),
                 "method": request.method,
-                "path": request.base_url._url,
+                "path": str(request.base_url)[:-1] + request.url.path if request.url.path else request.base_url,
```

Comment:
I think it'd be better to log scheme, host/domain, and path as separate entities. Path in the context of URL semantics really is just the `/trailing` portion after the domain (and my understanding of it would be that that doesn't include any query or fragments portions of the URL either).

Reply:
So we're on the same page, cause I'm confused.

Given a full URL: `https://somedomain.com/path/location/somewhere?foo=bar`

What are you saying was the intended value for the `path` param in the logging? `/path/location/somewhere` or `somedomain.com/path/location/somewhere` or `https://somedomain.com/path/location/somewhere` or the entire URL?

Reply:
Got it. I do think we should do what we have to (what you have in this PR) for backwards compat to ensure that it's consistently the same thing (full URL sans query param and fragment).

But... that's definitely not what I think of when I hear path :/ (I'd think path would mean `/path/location/somewhere`).

Reply:
Yeah, I get that-- if you want I can get a second opinion from Ian or Jag to make sure I'm on the right track here?

Reply:
Something I just realized -- what does the Flask plugin do? We need to ensure we're consistent with that one as well

Reply:
@Quidge The Flask plugin outputs the full URL without query params for the path (i.e. what the FastAPI one will do after this PR). Thanks for asking about that, I hadn't thought to check until now.

Reply:
Sounds good, thanks. If you're comfortable approving, I can release a new bugfix version.